src/pages/Search.tsx

```diff
@@ -11,23 +11,61 @@ import AddAgentCard from 'src/components/AddAgentCard'
 import { useAuth } from 'src/contexts/auth/AuthContext'
 import { useLogin } from 'src/contexts/LoginUIProvider/LoginUIContext'
 import ConfirmationModal from 'src/components/ConfirmationModal'
+import { Spinner } from 'react-bootstrap'
 
 export default function Search() {
   const {
     searchData: { data, meta },
     searchParam,
+    getData,
   } = useLoaderData() as SearchLoaderReturn
   const submit = useSubmit()
   const navigate = useNavigate()
   const { t } = useTranslation()
   const { user } = useAuth()
   const { openLogin } = useLogin()
   const [showConfirmModal, setShowConfirmModal] = useState(false)
+  const [items, setItems] = useState(data)
+  const [page, setPage] = useState(meta.page)
+  const [isLoading, setIsLoading] = useState(false)
+  const observerTarget = useRef(null)
+
+  const getMore = useCallback(async () => {
+    if (page >= meta.totalPages - 1 || isLoading) return
+    setIsLoading(true)
+    const newData = await getData(page + 1)
+    setItems([...items, ...newData.data])
+    setPage(newData.meta.page)
+
+    setIsLoading(false)
+  }, [page, meta, items, getData, isLoading])
+
+  useEffect(() => {
+    const currentObserverTarget = observerTarget.current
```

Comment:
Added this variable per "The ref value 'observerTarget.current' will likely have changed by the time this effect cleanup function runs. If this ref points to a node rendered by React, copy 'observerTarget.current' to a variable inside the effect, and use that variable in the cleanup function.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)"

Reply:
Rad